### PR TITLE
FIX: C++ std::tuple constexpr initial list on old compiler

### PR DESCRIPTION
**Pull request by feishengfei**
Fixing std::tuple initial list on old compiler.

compile error log on old compiler
```
/home/sietium/sarah/assimp/test/unit/utglTF2ImportExport.cpp: In member function ‘virtual void utglTF2ImportExport_wrongTypes_Test::TestBody()’:
/home/sietium/sarah/assimp/test/unit/utglTF2ImportExport.cpp:762:5: error: converting to ‘std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {const char (&)[32], const char (&)[6], const char (&)[11], const char (&)[10]}; <template-parameter-2-2> = void; _Elements = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’
     };
     ^
/home/sietium/sarah/assimp/test/unit/utglTF2ImportExport.cpp:762:5: error: converting to ‘std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {const char (&)[33], const char (&)[7], const char (&)[5], const char (&)[10]}; <template-parameter-2-2> = void; _Elements = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’
/home/sietium/sarah/assimp/test/unit/utglTF2ImportExport.cpp:762:5: error: converting to ‘std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {const char (&)[31], const char (&)[5], const char (&)[6], const char (&)[13]}; <template-parameter-2-2> = void; _Elements = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’
/home/sietium/sarah/assimp/test/unit/utglTF2ImportExport.cpp:762:5: error: converting to ‘std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {const char (&)[33], const char (&)[7], const char (&)[6], const char (&)[13]}; <template-parameter-2-2> = void; _Elements = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’
/home/sietium/sarah/assimp/test/unit/utglTF2ImportExport.cpp:762:5: error: converting to ‘std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {const char (&)[33], const char (&)[7], const char (&)[21], const char (&)[13]}; <template-parameter-2-2> = void; _Elements = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’
/home/sietium/sarah/assimp/test/unit/utglTF2ImportExport.cpp:762:5: error: converting to ‘std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {const char (&)[36], const char (&)[7], const char (&)[22], const char (&)[13]}; <template-parameter-2-2> = void; _Elements = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’
test/CMakeFiles/unit.dir/build.make:635: recipe for target 'test/CMakeFiles/unit.dir/unit/utglTF2ImportExport.cpp.o' failed
make[2]: *** [test/CMakeFiles/unit.dir/unit/utglTF2ImportExport.cpp.o] Error 1
make[2]: *** 正在等待未完成的任务....
CMakeFiles/Makefile2:229: recipe for target 'test/CMakeFiles/unit.dir/all' failed
make[1]: *** [test/CMakeFiles/unit.dir/all] Error 2
Makefile:135: recipe for target 'all' failed
make: *** [all] Error 2



sietium@sietium-RG-CT7600-2000-V2:~/sarah/assimp$ gcc --version
gcc (Ubuntu/Linaro 5.4.0-6kord1~16.04.12) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

sietium@sietium-RG-CT7600-2000-V2:~/sarah/assimp$ g++ --version
g++ (Ubuntu/Linaro 5.4.0-6kord1~16.04.12) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

### CODE REVIEW

File: test/unit/utglTF2ImportExport.cpp [hunk 1]
```diff
@@ -753,12 +753,21 @@ TEST_F(utglTF2ImportExport, wrongTypes) {
     // Deliberately broken version of the BoxTextured.gltf asset.
     using tup_T = std::tuple<std::string, std::string, std::string, std::string>;
     std::vector<tup_T> wrongTypes = {
```
Comment (Skylion007) [inline test/unit/utglTF2ImportExport.cpp:755]: Out of curiosity, if you make an alias for `std::vector<tup_T>`, does this also solve the compiler error?

Comment (feishengfei) [reply test/unit/utglTF2ImportExport.cpp:755]: Thanks for your suggestion. I will try it and feedback later.

**Comment by feishengfei** [comment]
@kimkulling your fix looks more concise and neat, but after merge I got following compile error:


```
[ 76%] Building CXX object test/CMakeFiles/unit.dir/unit/ImportExport/Assxml/utAssxmlImportExport.cpp.o
[ 76%] Building CXX object test/CMakeFiles/unit.dir/unit/utLWSImportExport.cpp.o
[ 76%] Building CXX object test/CMakeFiles/unit.dir/unit/utLWOImportExport.cpp.o
[ 77%] Building CXX object test/CMakeFiles/unit.dir/unit/utSMDImportExport.cpp.o
[ 77%] Building CXX object test/CMakeFiles/unit.dir/unit/utglTFImportExport.cpp.o
[ 77%] Building CXX object test/CMakeFiles/unit.dir/unit/utglTF2ImportExport.cpp.o
[ 78%] Building CXX object test/CMakeFiles/unit.dir/unit/utHMPImportExport.cpp.o
[ 78%] Building CXX object test/CMakeFiles/unit.dir/unit/utIFCImportExport.cpp.o
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:761:84: error: macro "TUPLE" passed 4 arguments, but takes just 1
  761 |         TUPLE("/glTF2/wrongTypes/badArray.gltf", "array", "primitives", "meshes[0]"),
      |                                                                                    ^
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:759: note: macro "TUPLE" defined here
  759 |     #define TUPLE(x) tup_T(x)
      | 
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:762:80: error: macro "TUPLE" passed 4 arguments, but takes just 1
  762 |         TUPLE("/glTF2/wrongTypes/badString.gltf", "string", "name", "scenes[0]"),
      |                                                                                ^
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:759: note: macro "TUPLE" defined here
  759 |     #define TUPLE(x) tup_T(x)
      | 
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:763:80: error: macro "TUPLE" passed 4 arguments, but takes just 1
  763 |         TUPLE("/glTF2/wrongTypes/badUint.gltf", "uint", "index", "materials[0]"),
      |                                                                                ^
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:759: note: macro "TUPLE" defined here
  759 |     #define TUPLE(x) tup_T(x)
      | 
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:764:84: error: macro "TUPLE" passed 4 arguments, but takes just 1
  764 |         TUPLE("/glTF2/wrongTypes/badNumber.gltf", "number", "scale", "materials[0]"),
      |                                                                                    ^
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:759: note: macro "TUPLE" defined here
  759 |     #define TUPLE(x) tup_T(x)
      | 
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:765:99: error: macro "TUPLE" passed 4 arguments, but takes just 1
  765 |         TUPLE("/glTF2/wrongTypes/badObject.gltf", "object", "pbrMetallicRoughness", "materials[0]"),
      |                                                                                                   ^
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:759: note: macro "TUPLE" defined here
  759 |     #define TUPLE(x) tup_T(x)
      | 
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:766:103: error: macro "TUPLE" passed 4 arguments, but takes just 1
  766 |         TUPLE("/glTF2/wrongTypes/badExtension.gltf", "object", "KHR_texture_transform", "materials[0]")
      |                                                                                                       ^
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:759: note: macro "TUPLE" defined here
  759 |     #define TUPLE(x) tup_T(x)
      | 
[ 78%] Building CXX object test/CMakeFiles/unit.dir/unit/utFBXImporterExporter.cpp.o
[ 78%] Building CXX object test/CMakeFiles/unit.dir/unit/utImporter.cpp.o
[ 79%] Building CXX object test/CMakeFiles/unit.dir/unit/ImportExport/utExporter.cpp.o
[ 79%] Building CXX object test/CMakeFiles/unit.dir/unit/ut3DImportExport.cpp.o
[ 79%] Building CXX object test/CMakeFiles/unit.dir/unit/ut3DSImportExport.cpp.o
[ 80%] Building CXX object test/CMakeFiles/unit.dir/unit/utACImportExport.cpp.o
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp: In member function ‘virtual void utglTF2ImportExport_wrongTypes_Test::TestBody()’:
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:761:9: error: ‘TUPLE’ was not declared in this scope
  761 |         TUPLE("/glTF2/wrongTypes/badArray.gltf", "array", "primitives", "meshes[0]"),
      |         ^~~~~
/home/felix/github/assimp/test/unit/utglTF2ImportExport.cpp:768:5: error: could not convert ‘{<expression error>, TUPLE, TUPLE, TUPLE, TUPLE, TUPLE}’ from ‘<brace-enclosed initializer list>’ to ‘std::vector<std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >’
  768 |     };
      |     ^
      |     |
      |     <brace-enclosed initializer list>
[ 80%] Building CXX object test/CMakeFiles/unit.dir/unit/utAMFImportExport.cpp.o
[ 80%] Building CXX object test/CMakeFiles/unit.dir/unit/utASEImportExport.cpp.o
[ 80%] Building CXX object test/CMakeFiles/unit.dir/unit/utD3MFImportExport.cpp.o
[ 81%] Building CXX object test/CMakeFiles/unit.dir/unit/utQ3DImportExport.cpp.o
[ 81%] Building CXX object test/CMakeFiles/unit.dir/unit/utSTLImportExport.cpp.o
gmake[2]: *** [test/CMakeFiles/unit.dir/build.make:678: test/CMakeFiles/unit.dir/unit/utglTF2ImportExport.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:186: test/CMakeFiles/unit.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
cmake --build build -- -j `nproc` V=s 2>&1  682.60s user 54.76s system 980% cpu 1:15.23 total
tee make.log  0.00s user 0.03s system 0% cpu 1:15.23 total
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
~/github/assimp (master) » gcc --version                                                                                                                                                               felix@earth
gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
~/github/assimp (master) » g++ --version                                                                                                                                                               felix@earth
g++ (Ubuntu 11.2.0-19ubuntu1) 11.2.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
~/github/assimp (master) »   
```

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Review by Skylion007 — COMMENTED** [review]
Not exactly what I had in mind, but if it works.

**Pull request merged**